source/client/output_formatter_impl.h

```diff
@@ -69,11 +69,11 @@ class FortioOutputFormatterImpl : public OutputFormatterImpl {
    *
    * @param result a single Nighthawk result, preferably the global result
    * @param counter_name the name of the counter to return
-   * @return the corresponding counter
-   * @throws NighthawkException if counter with given name is not found
+   * @param callback The callback to run iff a counter is found, passing the counter as an argument.
+   * @return True iff a counter was found.
    */
-  const nighthawk::client::Counter& getCounterByName(const nighthawk::client::Result& result,
-                                                     absl::string_view counter_name) const;
+  bool visitCounter(const nighthawk::client::Result& result, absl::string_view counter_name,
```

Comment:
Would it simplify the calling code to have a parameter that if set to true asks visitCounter to throw an error if the counter is not found? (Not sure if that would be mixing up concerns.)

Reply:
This doesn't seem to be needed, given that we only throw an error in the absence of one particular counter.